addons/website_sale_autocomplete/controllers/main.py

```diff
@@ -0,0 +1,151 @@
+        except TimeoutError:
+            return []
+        results = results['predictions']
+
+        # Convert google specific format to standard format.
+        return [{
+            'formatted_address': result['description'],
+            'google_place_id': result['place_id'],
+        } for result in results]
+
+    def _perform_complete_place_search(self, address, google_place_id=None, language_code=None, session_uuid=None):
+        params = {
+            'key': self._get_key(),
+            'place_id': google_place_id,
+            'fields': 'address_component,adr_address'
+        }
+
+        if language_code:
+            params['language'] = language_code
+        if session_uuid:
+            params['sessiontoken'] = session_uuid
+
+        try:
+            results = requests.get(f'{GOOGLE_PLACES_ENDPOINT}/details/json', params=params, timeout=TIMEOUT).json()
+        except TimeoutError:
+            return {'address': None}
+
+        html_address = results['result']['adr_address']
+        results = results['result']['address_components']  # Get rid of useless extra data
+
+        # Keep only the first type from the list of types
+        for res in results:
+            res['type'] = res.pop('types')[0]
+
+        # Sort the result by their priority.
+        results.sort(key=lambda r: FIELDS_PRIORITY.index(r['type']) if r['type'] in FIELDS_PRIORITY else 100)
+
+        standard_address = self._translate_google_to_standard(results)
+
+        if 'number' not in standard_address:
+            standard_address['number'] = self._guess_number_from_input(address, standard_address)
+            standard_address['formatted_street_number'] = f'{standard_address["number"]} {standard_address.get("street", "")}'
+        else:
+            formatted_from_html = html2plaintext(html_address.split(',')[0])
+            formatted_manually = f'{standard_address["number"]} {standard_address.get("street", "")}'
+            # Sometimes, the google api sends back abbreviated data :
+            # "52 High Road Street" becomes "52 HR St" for example. We usually take the result from google, but if it's an abbreviation, take our guess instead.
+            if len(formatted_from_html) >= len(formatted_manually):
+                standard_address['formatted_street_number'] = formatted_from_html
+            else:
+                standard_address['formatted_street_number'] = formatted_manually
+        return {
+            'address': standard_address
+        }
+
+    @http.route('/autocomplete/address', methods=['POST'], type='json', auth='public', website=True)
+    def _autocomplete_address(self, partial_address):
+
+        return {
+            'template': request.env['ir.qweb']._render('website_sale_autocomplete.address_autocomplete', {
+                'results': self._perform_place_search(partial_address, session_uuid=request.session.sid)
+            })
+        }
```

Comment:
Why is the response an object with a single key?

And why is said key called "template" when it doesn't seem to be any sort of template, but rather a rendered document fragment?